pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -262,8 +296,146 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 	return preds
 }
 
-func (s *cyclonedxParser) getPackageElement(elementID string) []*model.PkgInputSpec {
-	if packNode, ok := s.packagePackages[string(elementID)]; ok {
+func (c *cyclonedxParser) getVulnerabilities(ctx context.Context, cdxBom *cdx.BOM) *vulnData {
+	if cdxBom.Vulnerabilities == nil {
+		return nil
+	}
+
+	var vInfo vulnData
+	var status model.VexStatus
+	var justification model.VexJustification
+	var publishedTime time.Time
+	layout := "2006-01-02T15:04:05.000Z"
+	for _, vulnerability := range *c.cdxBom.Vulnerabilities {
+		vuln, err := asmhelpers.CreateVulnInput(vulnerability.ID)
+		if err != nil {
+			return nil
+		}
+
+		if vexStatus, ok := vexStatusMap[vulnerability.Analysis.State]; ok {
+			status = vexStatus
+		}
+
+		if vexJustification, ok := justificationsMap[vulnerability.Analysis.Justification]; ok {
+			justification = vexJustification
+		}
+
+		time, err := time.Parse(layout, vulnerability.Published)
+		if err == nil {
+			publishedTime = time
+		}
+
+		vd := model.VexStatementInputSpec{
+			Status:           status,
+			VexJustification: justification,
+			KnownSince:       publishedTime,
+			Statement:        vulnerability.Analysis.Detail,
+			StatusNotes:      fmt.Sprintf("%s:%s", string(vulnerability.Analysis.State), string(vulnerability.Analysis.Justification)),
+		}
+
+		for _, affect := range *vulnerability.Affects {
+			vi := c.getAffectedPackages(ctx, vuln, vd, affect)
+			if vi == nil {
+				continue
+			}
+			vInfo.Vex = append(vInfo.Vex, *vi...)
+
+			for _, v := range *vi {
+				if status == model.VexStatusAffected || status == model.VexStatusUnderInvestigation {
+					cv := assembler.CertifyVulnIngest{
+						Vulnerability: vuln,
+						VulnData: &model.ScanMetadataInput{
+							TimeScanned: publishedTime,
+						},
+						Pkg: v.Pkg,
+					}
+					vInfo.CertifyVuln = append(vInfo.CertifyVuln, cv)
+				}
+			}
+		}
+
+		for _, vulnRating := range *vulnerability.Ratings {
+			vm := assembler.VulnMetadataIngest{
+				Vulnerability: vuln,
+				VulnMetadata: &model.VulnerabilityMetadataInputSpec{
+					ScoreType:  model.VulnerabilityScoreType(vulnRating.Method),
+					ScoreValue: *vulnRating.Score,
+					Timestamp:  publishedTime,
+				},
+			}
+			vInfo.VulnMetadata = append(vInfo.VulnMetadata, vm)
+		}
+	}
+
+	return &vInfo
+}
+
+// Get package name and range versions to create package input spec for the affected packages.
+func (c *cyclonedxParser) getAffectedPackages(ctx context.Context, vulnInput *model.VulnerabilityInputSpec, vexData model.VexStatementInputSpec, affectsObj cdx.Affects) *[]assembler.VexIngest {
+	logger := logging.FromContext(ctx)
+	var pkgRef string
+	// TODO: retrieve purl from metadata if present - https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go#L76
```

Comment:
this should be handled by the above? Obtaining purl from metadata?

Reply:
Wasn't sure how true this would be incase of a vex bom, will the metadata be always present?